### PR TITLE
`azurerm_container_app_resource`: fix `TestAccContainerAppResource_completeWithVNet` and `TestAccContainerAppResource_completeTcpExposedPort`

### PR DESCRIPTION
**Pull request by jiaweitao001**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave "+1" or "me too" comments, they generate extra noise for PR followers and do not help prioritize for review

## Description

These 2 tests are failing of changes made to enable workload profile in container app resources. Removing delegation of subnet and workload profile in the test template to fix these tests.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [x] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [ ] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [ ] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [ ] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.
- [ ] (For changes that include a **state migration only**). I have manually tested the migration path between relevant versions of the provider.

## Testing 

- [x] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_container_app_resource` - fix `TestAccContainerAppResource_completeWithVNet` and `TestAccContainerAppResource_completeTcpExposedPort`

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

File: internal/services/containerapps/container_app_environment_resource_test.go [hunk 1]
```diff
@@ -455,6 +481,27 @@ resource "azurerm_subnet" "control" {
 `, r.template(data), data.RandomInteger)
 }
 
+func (r ContainerAppEnvironmentResource) templateVnetSubnetNotDelegated(data acceptance.TestData) string {
```
Comment (jackofallops) [inline internal/services/containerapps/container_app_environment_resource_test.go:484]: Subnets are supposed to need delegation for Environments to use them according to the docs afaik, what's the reasoning for this needing it removed?

**Review by jackofallops — COMMENTED** [review]
Thanks @jiaweitao001 - this looks mostly fine, just an operational question that needs some additional information below if you can take a look ?

File: internal/services/containerapps/container_app_environment_resource_test.go (hunk 1 above)
Comment (jiaweitao001) [reply internal/services/containerapps/container_app_environment_resource_test.go:484]: A delegated subnet cannot be used for creating container app environment according to the API response. Here's the error message if it's delegated:
```
Status: "Failed"
Code: "ManagedEnvironmentSubnetIsDelegated"
Message: "AgentPoolProfile subnet with id xxxx cannot be used as it's a delegated subnet. Please check https://aka.ms/adv-network-prerequest for more details.\r\nStatus: 400 (Bad Request)\r\nErrorCode: SubnetIsDelegated\r\n\r\nContent:\r\n{\n  \"code\": \"SubnetIsDelegated\",\n  
```

**Review by katbyte — APPROVED** [review]
LGTM 🌳

**Pull request merged**